ProcessLib/Process.h

```diff
@@ -12,35 +12,187 @@
+	/// parsing the configuration.
+	void setLinearSolveOptions(
+	    boost::optional<const BaseLib::ConfigTree&> config)
+	{
+		if (config)
+		{
+			_linear_solver_options.reset(
+			    new BaseLib::ConfigTree(*config));
+		}
+	}
+
+	/// Creates global matrix, rhs and solution vectors, and the linear solver.
+	void createLinearSolver(
+	    AssemblerLib::LocalToGlobalIndexMap const& local_to_global_index_map,
+	    std::string const solver_name)
+	{
+		DBUG("Allocate global matrix, vectors, and linear solver.");
+#ifdef USE_PETSC
+		MathLib::PETScMatrixOption mat_opt;
+		const MeshLib::NodePartitionedMesh& pmesh =
+		    static_cast<const MeshLib::NodePartitionedMesh&>(_mesh);
+		mat_opt.d_nz = pmesh.getMaximumNConnectedNodesToNode();
+		mat_opt.o_nz = mat_opt.d_nz;
+		const std::size_t num_unknowns =
+		    local_to_global_index_map.dofSizeGlobal();
+		_A.reset(_global_setup.createMatrix(num_unknowns, mat_opt));
+#else
+		const std::size_t num_unknowns = local_to_global_index_map.dofSize();
+		_A.reset(_global_setup.createMatrix(num_unknowns));
+#endif
+		_x.reset(_global_setup.createVector(num_unknowns));
+		_rhs.reset(_global_setup.createVector(num_unknowns));
+		_linear_solver.reset(new typename GlobalSetup::LinearSolver(
+		    *(_A), solver_name, _linear_solver_options.get()));
+	}
+
+	/// Computes and stores global matrix' sparsity pattern from given
+	/// DOF-table.
+	void computeSparsityPattern(
+	    AssemblerLib::LocalToGlobalIndexMap const& local_to_global_index_map)
+	{
+		_sparsity_pattern = std::move(AssemblerLib::computeSparsityPattern(
+		    local_to_global_index_map, _mesh));
+	}
+
+
+	/// Sets the initial condition values in the solution vector x for a given
+	/// process variable and component.
+	void setInitialConditions(ProcessVariable const& variable,
+	                          int const component_id)
+	{
+		std::size_t const n = _mesh.getNNodes();
+		for (std::size_t i = 0; i < n; ++i)
+		{
+			MeshLib::Location const l(_mesh.getID(),
+			                          MeshLib::MeshItemType::Node, i);
+			auto const global_index =  // 0 is the component id.
+			    std::abs(_local_to_global_index_map->getGlobalIndex(
+			        l, component_id));
+			_x->set(global_index,
+			        variable.getInitialConditionValue(*_mesh.getNode(i)));
+		}
+	}
 
 protected:
-    MeshLib::Mesh& _mesh;
+	MeshLib::Mesh& _mesh;
+	std::vector<MeshLib::MeshSubsets*> _all_mesh_subsets;
+
+	GlobalSetup _global_setup;
+
+	using GlobalAssembler =
+	    AssemblerLib::VectorMatrixAssembler<typename GlobalSetup::MatrixType,
+	                                        typename GlobalSetup::VectorType>;
+
+	std::unique_ptr<GlobalAssembler> _global_assembler;
+
+	std::unique_ptr<AssemblerLib::LocalToGlobalIndexMap>
+	    _local_to_global_index_map;
+
+	std::unique_ptr<BaseLib::ConfigTree> _linear_solver_options;
+	std::unique_ptr<typename GlobalSetup::LinearSolver> _linear_solver;
+
+	std::unique_ptr<typename GlobalSetup::MatrixType> _A;
```

Comment:
While this is not implemented yet, the current changes don't prohibit it. We shall see how it is possible when there are multiple weakly coupled processes, which shall not happen before March or so.